asserts/database.go

```diff
@@ -89,6 +89,18 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// SupersededRevisionError is the error returned when an assertion is added
```

Comment:
Sorry, that's not appropriate for all cases. Some assertions will be built at the client side, and for those version skews are not so hard to observe.

At the same time, going back to your earlier point, if we can have a matching revision being an error, that's not quite a superseded revision. It's more like a conflict.

So, here is a suggested phrasing, making the revisions public too after the comments above, and going back to a general error type that can hopefully address all cases we'll need it for:
```
// RevisionError indicates a revision improperly used for an operation.
type RevisionError struct {
        Used, Current int
}

func (e *RevisionError) Error() string {
        if e.Used == 0 || e.Current == 0 {
                return fmt.Errorf("assertion revision %d is unknown", e.Used)
        }
        if e.Used == e.Current {
                return fmt.Sprintf("revision %d is already the current revision", e.Used)
        }
        if e.Used < e.Current {
                return fmt.Sprintf("revision %d is older than current revision %d", e.Used, e.Current)
        }
        return fmt.Sprintf("revision %d is more recent than current revision %d", e.Used, e.Current)
}